### PR TITLE
Feature/11154 create fss mock api for querying & download for enc files

### PR DESCRIPTION
**Pull request by ganpat-tech**
Created FSS mock api end points for Querying and Download for ENC files.

### CODE REVIEW

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Services/FileShareService.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        public SearchBatchResponse GetBatches(string filter)
+        {
+            if (!string.IsNullOrEmpty(filter))
+            {
+                return FileHelper.ReadJsonFile<SearchBatchResponse>(fileShareServiceConfiguration.Value.FileDirectoryPath + fileShareServiceConfiguration.Value.ScsResponseFile); 
+            }
+            return null;
+        }
+
+        public byte[] GetENCFilePath(string filesName)
```
Comment (crdandoukho) [inline UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Services/FileShareService.cs:28]: `GetENCFilePath()` will always return the _first_ TXT or 000 file in the directory - only the extension of the `filesName` parameter is used by the file selection logic.

Please ignore this comment if this is the intended behaviour

Comment (ganpat-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Services/FileShareService.cs]: Thanks for highlighting it, making the changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 2]
```diff
@@ -16,5 +32,29 @@ public IActionResult CreateBatch(string correlationId)
...
+                }
+            }
+            return BadRequest();
+        }
+
+        [HttpGet]
+        [Route("/batch/{batchId}/files/{filesName}")]
+        public FileResult DownloadENCFile(string filesName)
+        {
+            byte[] bytes = fileShareService.GetEncFileData(filesName);
```
Comment (janvi13603) [inline UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs:55]: Code to download readme.txt file has to be added and also change name of method DownloadFile()

Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Did changes accordingly. Thanks!

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Services/FileShareService.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            if (filter.Contains("README.TXT", StringComparison.OrdinalIgnoreCase))
+            {
+                response.Entries.RemoveRange(1, response.Entries.Count - 1);
+            }
+            return response;
+        }
+
+        public byte[] GetFileData(string filesName)
+        {
+            string filePath, fileType = Path.GetExtension(filesName);
```
Comment (barrie-cooper) [inline UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Services/FileShareService.cs:32]: better to declare `filePath` where it is used

**Review by barrie-cooper — APPROVED** [review]
Looks good

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Services/FileShareService.cs (hunk 3 above)
Comment (Siddartha-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Services/FileShareService.cs]: Yes. Did changes accordingly. Thanks!

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**